src/app/saga.ts

```diff
@@ -318,21 +309,19 @@ function convertQueryToScreenParams(query: string) {
 }
 
 export function* handleDeepLink(action: OpenDeepLink) {
-  let { deepLink } = action
+  const { deepLink } = action
   const { isSecureOrigin } = action
   Logger.debug(TAG, 'Handling deep link', deepLink)
 
-  if (isWalletConnectDeepLink(deepLink)) {
-    yield* call(handleWalletConnectDeepLink, deepLink)
+  const walletAddress = yield* select(walletAddressSelector)
+  if (!walletAddress) {
+    Logger.error(TAG, 'No wallet address found in store. This should never happen.')
     return
   }
 
-  // Try resolve dynamic links
-  if (deepLink.startsWith(DYNAMIC_LINK_DOMAIN_URI_PREFIX)) {
-    const resolvedDynamicLink = yield* call(resolveDynamicLink, deepLink)
-    if (resolvedDynamicLink) {
-      deepLink = resolvedDynamicLink
-    }
+  if (isWalletConnectDeepLink(deepLink)) {
+    yield* call(handleWalletConnectDeepLink, deepLink)
+    return
   }
```

Comment:
Are we sure this isn't needed anymore?
Maybe this was needed for links scanned via the in-app QR scanner?
or when passed from our own notifications?

Reply:
i *think* that it's not needed anymore...

the QR scanner [saga](https://github.com/valora-inc/wallet/blob/0c4adb8166258c184f171011e94025971ab681ac/src/qrcode/utils.ts#L146-L174) does not trigger the deep link flow currently. if we want to be able to handle dynamic links through the scanner, we'll need to make a code change here anyway so i thought it was a safe change.

the "passed from our own notifications" scenario is interesting, but i didn't think it was a real scenario? the notifications (both homecards and push notifications) actually don't work well with dynamic links (i just tested) because[ it will try to open it in a browser](https://github.com/valora-inc/wallet/blob/main/src/app/saga.ts#L407). both flows end up there. even if it does work, i guess i'm wondering why someone internal would choose to add a dynamic link instead of a deep link to target an existing user. it seems like the end result is the same, but the dynamic link option is more work for the creator of the link. for all other cases, the dynamic link would come through the normal deep links handler.

i want to minimise the amount of dead code, and i'm fairly sure this block will not be used. but if we're unsure then there's no harm to put it back (this is the less risky option for sure)?

Reply:
Sounds good, thanks for checking everything, we should remove then 👍